CRM/Contribute/Tokens.php

```diff
@@ -194,7 +194,43 @@ public function evaluateToken(TokenRow $row, $entity, $field, $prefetch = NULL)
       $row->tokens($entity, $field, $fieldValue);
     }
     else {
-      $row->dbToken($entity, $field, 'CRM_Contribute_BAO_Contribution', $field, $fieldValue);
+      // This section is copied & pasted & can be cleaned up later.
+      // most if probably already done in this functions
+      $tokenEntity = $entity;
+      $tokenField = $field;
+      $baoName = $this->getBAOName();
+      $baoField = $field;
+      if ($fieldValue === NULL || $fieldValue === '') {
+        return $row->tokens($tokenEntity, $tokenField, '');
+      }
+
+      $fields = $this->getFieldMetadata();
+      if (!empty($fields[$baoField]['pseudoconstant'])) {
+        $options = $baoName::buildOptions($baoField, 'get');
```

Comment:
The if-else further up exists mostly to circumvent the code ever reaching this so once I am comfortable with the test cover I will remove